### PR TITLE
Cookie parsing in context

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request introduces several enhancements to the authentication and context creation mechanisms, along with updates to token expiration configurations and testing improvements. Key changes include the addition of cookie parsing in the GraphQL context, dynamic token expiration settings, and expanded test coverage for edge cases in context and token handling.

### Enhancements to context creation:

* [`src/context.ts`](diffhunk://#diff-0cd594dddd6a9f2e3e26afebbb92716434437d7001c2416252aa7531e7f2b8d4R3): Added support for parsing cookies from request headers using the `cookie` library and included a `response` object with headers for each context. Improved error handling for JWT token verification to gracefully handle invalid or expired tokens. [[1]](diffhunk://#diff-0cd594dddd6a9f2e3e26afebbb92716434437d7001c2416252aa7531e7f2b8d4R3) [[2]](diffhunk://#diff-0cd594dddd6a9f2e3e26afebbb92716434437d7001c2416252aa7531e7f2b8d4R12-R53) [[3]](diffhunk://#diff-0cd594dddd6a9f2e3e26afebbb92716434437d7001c2416252aa7531e7f2b8d4L32-R91)

### Updates to token expiration configuration:

* [`src/config/env.ts`](diffhunk://#diff-0628e9bc652476c15d1024481f1eaebb8f1ca780b4ac24da3b2b20651662968aL16-R20): Introduced environment variables `ACCESS_TOKEN_EXPIRES_IN_SECONDS` and `REFRESH_TOKEN_EXPIRES_IN_DAYS` for dynamic token expiration settings.
* [`src/auth/token.ts`](diffhunk://#diff-1f2003d64fec054fd826b42de19417f5eb0639b76b6bcaf0e5a0a404937d13eaL12-R12): Updated access token and refresh token expiration logic to use the new environment variables, replacing hardcoded values. [[1]](diffhunk://#diff-1f2003d64fec054fd826b42de19417f5eb0639b76b6bcaf0e5a0a404937d13eaL12-R12) [[2]](diffhunk://#diff-1f2003d64fec054fd826b42de19417f5eb0639b76b6bcaf0e5a0a404937d13eaL50-R49) [[3]](diffhunk://#diff-1f2003d64fec054fd826b42de19417f5eb0639b76b6bcaf0e5a0a404937d13eaL88-R87)

### Expanded test coverage:

* [`src/__tests__/context.test.ts`](diffhunk://#diff-7e6b705394a71db8613c07b52b94e2582a3e4d9b6ae7aa73598ce3d239a6763cR1-R214): Added comprehensive tests for `createContext`, including scenarios for authorization headers, cookie parsing, response headers, and edge cases such as malformed cookies and database errors.
* [`src/auth/__tests__/token.test.ts`](diffhunk://#diff-97f6bfd19abb0af4a33c1bac200cace552cb4f0308ccd9b5b8246f93c5f5b869L38-R40): Updated tests for token utilities to reflect the new expiration settings and added test cases for edge scenarios. [[1]](diffhunk://#diff-97f6bfd19abb0af4a33c1bac200cace552cb4f0308ccd9b5b8246f93c5f5b869L38-R40) [[2]](diffhunk://#diff-97f6bfd19abb0af4a33c1bac200cace552cb4f0308ccd9b5b8246f93c5f5b869L120-R122) [[3]](diffhunk://#diff-97f6bfd19abb0af4a33c1bac200cace552cb4f0308ccd9b5b8246f93c5f5b869L199-R201)

### Dependency updates:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R43-R47): Added `@types/cookie` and `cookie` dependencies to support cookie parsing functionality.

### Testing improvements:

* [`src/testSetup.ts`](diffhunk://#diff-788f1913e46fa32eea4205cf6b83dd2576268b9a2063c3e07de9a0811f2b9914L4-R9): Refactored Prisma client mocking to ensure proper import and usage in tests.
* [`vitest.config.js`](diffhunk://#diff-bf00a51cbcbfbf0db010355fe55fe1d8417c1f188068e9d688e077820a398cb3L19): Removed `src/context.ts` from coverage exclusion to include context-related tests in coverage reports.

### CODE REVIEW

File: src/auth/token.ts [hunk 1]
```diff
@@ -47,7 +46,7 @@ export async function verifyGoogleToken(token: string) {
 
 export function generateAccessToken(payload: TokenPayload): string {
     return jwt.sign(payload, env.JWT_SECRET, {
-        expiresIn: ACCESS_TOKEN_EXPIRES_IN
+        expiresIn: env.ACCESS_TOKEN_EXPIRES_IN_SECONDS
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/auth/token.ts:49]: might be reading an old commit

File: src/__tests__/context.test.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            mockRequest.headers.set('authorization', 'Bearer   valid-token');
+            vi.mocked(verifyJWT).mockReturnValue({
+                userId: 'user-123',
+                email: 'test@example.com'
+            });
+            prismaMock.user.findUnique.mockResolvedValue(mockUser);
+
+            const context = await createContext(mockYogaContext);
+
+            expect(verifyJWT).toHaveBeenCalledWith('  valid-token');
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/__tests__/context.test.ts]: removing this test

File: src/__tests__/context.test.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        it('should handle case-insensitive Bearer prefix', async () => {
+            mockRequest.headers.set('authorization', 'bearer valid-token');
+            vi.mocked(verifyJWT).mockReturnValue({
+                userId: 'user-123',
+                email: 'test@example.com'
+            });
+
+            await createContext(mockYogaContext);
+
+            expect(verifyJWT).toHaveBeenCalledWith('bearer valid-token');
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/__tests__/context.test.ts:182]: updating implementation to check for proper format
